### PR TITLE
Add OPAM file as well as GitHub action CI job for Coq >= 8.4

### PR DESCRIPTION
**Pull request by aerabi**

### CODE REVIEW

File: meta.yml [hunk 1]
```diff
@@ -31,3 +32,19 @@ keywords:
   - name: type theory
 categories:
   - name: Mathematics/Logic/Type theory
+action: true
+supported_coq_versions:
+  - text: Coq 8.4 or higher
+    opam: '{(>= "8.7" & < "8.13~") | (= "dev")}'
```
Comment (anton-trunov) [inline meta.yml]:
```suggestion
    opam: '{(>= "8.4" & < "8.13~") | (= "dev")}'
```

**Review by Zimmi48 — APPROVED** [review]
Looks terrific!

The compatibility range is impressive. At some point, you'll probably benefit from dropping a few of the earlier versions. E.g., if you want to fix the warning "Declaring a scope implicitly is deprecated", you need Coq 8.10+ to use the `Declare Scope` command.

**Comment by aerabi** [comment]
Thanks, @Zimmi48. Let's keep the compatibility range wide up to the point we are forced to shorten it. :pray:

**Comment by aerabi** [comment]
Despite reading [this document](https://coq.inria.fr/opam-packaging.html), I still don't know how the packaging works. I tagged the current state of master, but still by doing `opam install coq-generic-environments`, I get the following message:

```
The following actions will be performed:
  ↘ downgrade coq                      8.12.0 to 8.10.2 [required by coq-generic-environments]
  ∗ install   conf-adwaita-icon-theme  1                [required by coqide]
  ∗ install   coq-generic-environments 8.10.0
  ↘ downgrade coqide                   8.12.0 to 8.10.2 [uses coq]
===== ∗ 2   ↘ 2 =====
```

**Comment by anton-trunov** [comment]
@aerabi You are supposed to make a PR to https://github.com/coq/opam-coq-archive and add a new subdirectory per each release here: https://github.com/coq/opam-coq-archive/tree/master/released/packages/coq-generic-environments.

**Pull request merged**